src/shogun/util/factory.h

```diff
@@ -333,5 +336,113 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+
+	// The function of those two class is to distinguish between string_feature
+	// and features_subset
+	class StringsFeatures;
+	class FeatureSubset;
+
+	template <typename TypeName, typename... Args>
+	std::shared_ptr<std::conditional_t<
+	    traits::is_any_of_v<TypeName, StringsFeatures, FeatureSubset>, Features,
+	    TypeName>>
+	create(Args... args)
```

Comment:
The SWIG don't handle with rvalue reference, so i don't use `perfect forwarding ` in here.

Reply:
I would hide the forwarding reference version with a macro guard.